source/common/http/http1/codec_impl.cc

```diff
@@ -1097,24 +1052,25 @@ Envoy::StatusOr<ConnectionImpl::HttpParserCode> ServerConnectionImpl::onHeadersC
     // with message complete. This allows upper layers to behave like HTTP/2 and prevents a proxy
     // scenario where the higher layers stream through and implicitly switch to chunked transfer
     // encoding because end stream with zero body length has not yet been indicated.
-    if (parser_.flags & F_CHUNKED ||
-        (parser_.content_length > 0 && parser_.content_length != ULLONG_MAX) || handling_upgrade_) {
+    if (parser_->flags() & parser_->flagsChunked() ||
+        (parser_->contentLength() > 0 && parser_->contentLength() != ULLONG_MAX) ||
```

Comment:
lazy check: is the uulong_max a "not set" value?    if so is that going to be consistent between parsers or should content length be optional and the legacy version return nullopt if uulong_max?

edit: ah yeah, I think this is seenContentLength() && >0 but I'd mildly prefer an optional and explicit checks or a todo to convert if you want to minimize churn here.

Reply:
love the optional! and yes, the UULONG_MAX was an artifact where seenContentLength() should have been used.